java-checks-test-sources/src/main/java/checks/spring/NonSingletonAutowiredInSingletonCheck.java

```diff
@@ -0,0 +1,139 @@
+package checks.spring;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.WebApplicationContext;
+
+public class NonSingletonAutowiredInSingletonCheck {
+
+  @Component
+  @Scope(value = WebApplicationContext.SCOPE_REQUEST, proxyMode = ScopedProxyMode.TARGET_CLASS)
+  public class RequestBean1 {
+    //...
+  }
+
+  public class SingletonBean {
+    @Autowired
+    private final RequestBean1 requestBean1; // Noncompliant, [[sc=19;ec=31]] {{Singleton beans should not auto-wire non-Singleton beans.}}
+
+    public SingletonBean() {
+      requestBean1 = null;
+    }
+  }
+
+  @Scope("prototype")
+  public class PrototypeBean1 {
+  }
+
+  public class SingletonBean1 {
+    @Autowired
+    private final PrototypeBean1 prototypeBean1; // Noncompliant, [[sc=19;ec=33]] {{Singleton beans should not auto-wire non-Singleton beans.}}
```

Comment:
It's enough to check the range for field injection once, it makes the tests easier to maintain if we don't check every time. It definitely does make sense to check the range again for other types of injection, like constructor injection. Again, only once though.

Same for the issue message, it's enough to only verify it once :).